src/main/java/spoon/support/sniper/internal/AbstractSourceFragmentContextCollection.java

```diff
@@ -52,6 +53,8 @@ public boolean knowsHowToPrint(PrinterEvent event) {
 				return false;
 			} else if (tpe.getType() == TokenType.IDENTIFIER) {
 				return findIndexOfNextChildTokenByType(TokenType.IDENTIFIER) >= 0;
+			} else if (tpe.getToken().equals(";") && childFragmentHasSpecifiedRoleInParent(CtRole.TRY_RESOURCE)) {
```

Comment:
With the suggested renaming, this becomes somewhat easier to interpret, I think:

```suggestion
			} else if (tpe.getToken().equals(";") && anyChildFragmentHasRole(CtRole.TRY_RESOURCE)) {
```